delta-app/src/main/java/io/cdap/delta/store/AssessmentServiceClient.java

```diff
@@ -38,71 +38,65 @@
 import java.time.temporal.ChronoUnit;
 import javax.annotation.Nullable;
 import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.MediaType;
 
 /**
  * Class to handle Api calls to System Service
  */
 public class AssessmentServiceClient {
 
-  ServiceDiscoverer serviceDiscoverer;
+  private final ServiceDiscoverer serviceDiscoverer;
+  private RetryPolicy retryPolicy;
+  private static final Gson GSON = new Gson();
 
   public AssessmentServiceClient(ServiceDiscoverer context) {
     this.serviceDiscoverer = context;
+    setUpRetryPolicy();
```

Comment:
nit: this looks like it's only called in the constructor, in which case it's a bit more readable just to create it here in the constructor